pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -547,15 +547,21 @@ public static boolean parseReturningKeyword(final char[] query, int offset) {
       return false;
     }
 
-    return (query[offset] | 32) == 'r'
+    /* test to make sure the word returning is not in the statement
+        we have to test for a space in front and in the end
+    */
+    return (query[offset - 1] == ' ')
```

Comment:
Agreed using offset-1 is very risky IMO, ideally we'd have some kind of look ahead parser to look for whitespace + 'r' before going into that code. I hadn't thought of newline...